src/ape/managers/_contractscache.py

```diff
@@ -270,13 +270,20 @@ def _delete_proxy(self, address: AddressType):
     def __contains__(self, address: AddressType) -> bool:
         return self.get(address) is not None
 
-    def cache_deployment(self, contract_instance: ContractInstance):
+    def cache_deployment(
+        self,
+        contract_instance: ContractInstance,
+        proxy_info: Optional[Union[ProxyInfoAPI, Literal["skip"]]] = None,
+    ):
         """
         Cache the given contract instance's type and deployment information.
 
         Args:
             contract_instance (:class:`~ape.contracts.base.ContractInstance`): The contract
               to cache.
+            proxy_info (Optional[Union[ProxyInfoAPI, Literal["skip"]]]): Either pass in the proxy
+              info, if it is known, to avoid the potentially expensive look-up, or pass the
+              keyword "skip" to skip detecting proxies for this contract.
```

Comment:
Might be better to just have 2 options here, one for proxy info and one for `skip_proxy_check=True`